### DISCUSSION TITLE
Working with mermaid in typst

### DISCUSSION THREAD

**Discussion by drbartling** [typst/typst#3090] [Q&A] [answered]
There's currently an open issue for rendering text in some SVGs: https://github.com/typst/typst/issues/1421
What are some of the workarounds folks have used to deal with this?

I think the main issue is that if I use PNG or other raster formats, the diagram gets blocky or large.
But one workaround is to use the `--scale` option to increase the size of the image so it renders nicely at most zoom levels used for the PDF.

In the issue thread, @natexcvi mentioned turning off `htmlLabels` but that didn't work for me.
Is anyone able to get SVG diagrams out of Mermaid with text rendered appropriately in typst?

**Comment by LaurenzV** [comment, marked as answer]
It seems like for now the best approach is to export it to PDF and then convert it to SVG (for example using `mootool`). For example:
 
First run (using [the Mermaid CLI](https://github.com/mermaid-js/mermaid-cli):
`mmdc -i in.mmd -o out.pdf`

And then run (for example using [mutool](https://www.mankier.com/1/mutool) which in my experience works best, but you can just use any other pdf-to-svg utility as well):
`mutool draw -o out.svg out.pdf`

And then you can just use the resulting SVG!

**Reply by drbartling** [reply]
2 things to note:

1. `mmdc -i in.mmd -o out.pdf --pdfFit` Will fit the PDF file to the image, which will help with layout when using the SVG we create next.
2. `mutool draw -o out.svg out.pdf` will create out1.svg since mutool creates a new file for each page.  In automation, I follow this up with `mv out1.svg out.svg` so that I don't leak that detail to later steps.

Otherwise, it's a good work-around.  Thanks for the help!

**Comment by tyoc213** [comment]
Why not use the direct output? `mmdc -i input.mmd -o output.svg`???

**Reply by YDX-2147483647** [reply]
> There's currently an open issue for rendering text in some SVGs: https://github.com/typst/typst/issues/1421
